### PR TITLE
TM-4916-Bid-Audit

### PR DESCRIPTION
**Pull request by cagbakwu**
https://metaphase.atlassian.net/browse/TM-4916

-Created new Admin page called Audit Bid.
-Added redux for submitting and deleting changes.
-Added 3 tables for cards with data.
-Added edit mode and read mode for tabs.

Screenshot 2023-09-26 at 8 14 29 AM
Screenshot 2023-09-26 at 8 14 51 AM
Screenshot 2023-09-26 at 8 14 57 AM
Screenshot 2023-09-26 at 8 15 13 AM

### CODE REVIEW

File: src/Components/AdministratorPage/bidAudit/BidAudit.jsx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+          <p className="add-at">
+            <FA name="plus" />
+            <Link
+              to="#"
+              onClick={onAddClick}
+            >
+              Create New Audit Cycle
+            </Link>
+          </p>
+          {dummyPositionDetails[2]?.bidAudit.map(k => (
```
Comment (cagbakwu) [inline src/Components/AdministratorPage/bidAudit/BidAudit.jsx:214]: Dummy data could possibly be written differently.

File: src/actions/bidAudit.js [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+  UPDATE_BID_AUDIT_ERROR_TITLE,
+  UPDATE_BID_AUDIT_SUCCESS,
+  UPDATE_BID_AUDIT_SUCCESS_TITLE,
+} from 'Constants/SystemMessages';
+import { batch } from 'react-redux';
+import api from '../api';
+import { toastError, toastSuccess } from './toast';
+
+// This could be arranged better, but for now, this is the data structure
+const dummyData = [
```
Comment (cagbakwu) [inline src/actions/bidAudit.js:16]: Dummy data could possibly be written differently.

**Comment by M00staff** [comment]
when i try to run this locally i get this slew of errors
image

also, there are conflicts & failing tests - please fix 

.. did you even check to see if this works?

File: src/Components/AdministratorPage/bidAudit/BidAudit.jsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  const genericFiltersIsLoading = useSelector(state => state.filtersIsLoading);
+  const genericFilters = useSelector(state => state.filters);
+
+  const [assignmentCycles, setAssignmentCycles] = useState(bidAuditCycles?.assignmentCycles || []);
+  const [auditNumber, setAuditNumber] = useState(bidAuditCycles?.auditNumber || []);
+  const [auditDescription, setAuditDescription] = useState(bidAuditCycles?.auditDescription || []);
+  const [postByDate, setPostByDate] = useState();
+  const [clearFilters, setClearFilters] = useState(false);
+
+  const genericFilters$ = get(genericFilters, 'filters') || [];
```
[the remark this answers is not in the document]
Comment (M00staff) [reply src/Components/AdministratorPage/bidAudit/BidAudit.jsx:36]: Don't be sorry - be better.

**Comment by elizabeth-jimenez** [comment]
Moose's comment above makes me nervous for this PR - so i put a Do Not Merge on it, just until we get the emergency release one underway - if it gets approved, I can deploy the branch to dev1 tho, for the demo tomorrow so no lose

**Comment by elizabeth-jimenez** [comment]
> Moose's comment above makes me nervous for this PR - so i put a Do Not Merge on it, just until we get the emergency release one underway - if it gets approved, I can deploy the branch to dev1 tho, for the demo tomorrow so no lose

'lose' tho 😩

**Comment by cagbakwu** [comment]
> Moose's comment above makes me nervous for this PR - so i put a Do Not Merge on it, just until we get the emergency release one underway - if it gets approved, I can deploy the branch to dev1 tho, for the demo tomorrow so no lose

@elizabeth-jimenez  ok no worries, its working fine locally for me but I can see what the potential issue is.

**Comment by cagbakwu** [comment]
@M00staff there was a folder capitalization that caused that error. Its resolved now and running. Ill check if there are any other issues while its being reviewed.

File: src/Components/AdministratorPage/bidAudit/BidAudit.jsx [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+            <div className="filterby-label">Filter by:</div>
+            <div className="filterby-clear">
+              {clearFilters &&
+                <button
+                  className="unstyled-button"
+                  onClick={resetFilters}
+                  disabled={disableSearch}
+                >
+                  <FA name="times" />
+                  Clear Filters
```
Comment (M00staff) [inline src/Components/AdministratorPage/bidAudit/BidAudit.jsx:132]: something is wrong with the clear filters logic - its always active

image

**Comment by cagbakwu** [comment]
Added some hardcoded data for now for the drop downs in bid audit. Will remove after demo.

**Comment by M00staff** [comment]
I think you may have misunderstood some of the A/C's
image

I think the implication here ^^ is that the button opens up a modal to create a new Audit, with those dropdowns

But, instead you added filters - i might be off and will double check - but i do think that's the case

image

**Comment by M00staff** [comment]
I'm going to see if i can fix these issues before the demo - but listing them here for my own reference

image

image

Edit button shows up twice on At Grades tab

**Comment by cagbakwu** [comment]
> I'm going to see if i can fix these issues before the demo - but listing them here for my own reference
> 
> image image
> Edit button shows up twice on At Grades tab

Looks like thats the default edit at the top. The one for the actual selected card shows when you select the checkbox. May have to pass something into the tabbed component.

File: public/config/config.json [hunk 5]
```diff
@@ -35,9 +39,6 @@
     "register_checklist": true,
     "search_post_access": true,
     "settings": true,
-    "unassigned_filters": true,
-    "bidding_tool": true,
-    "assignment_notification": true,
-    "assignment_memo": true
+    "unassigned_filters": true
```
Comment (M00staff) [inline public/config/config.json:42]: looks like someone's ESLint auto format got this ... might wanna play with your settings so it doesn't format lines you haven't touched

Comment (M00staff) [reply public/config/config.json:42]: @musiqe31

File: src/Components/BidderPortfolio/BidderPortfolioStatCard/BidderPortfolioStatCard.jsx [hunk 6]
```diff
@@ -29,16 +29,14 @@ const BidderPortfolioStatCard = ({ userProfile, showEdit, classifications, viewT
   const languages = get(userProfile, 'current_assignment.position.language');
   const bidder = get(userProfile, 'shortened_name') || 'None listed';
   // This is the new key bidder_types. It returns a string of either 'cusp' or 'eligible'
-  const bidderType = get(userProfile, 'bidder_types') || null;
-  // eslint-disable-next-line no-unused-vars
+  const bidderType = 'eligible';
```
Comment (M00staff) [inline src/Components/BidderPortfolio/BidderPortfolioStatCard/BidderPortfolioStatCard.jsx:32]: hardcoding to demo - this is for the CDO D3 stuff

File: src/Components/BidderPortfolio/BidderPortfolioStatRow/BidderPortfolioStatRow.jsx [hunk 7]
```diff
@@ -29,16 +29,14 @@ const BidderPortfolioStatRow = ({ userProfile, showEdit, classifications, viewTy
   const languages = get(userProfile, 'current_assignment.position.language');
   const bidder = get(userProfile, 'shortened_name') || 'None listed';
   // This is the new key bidder_types. It returns a string of either 'cusp' or 'eligible'
-  const bidderType = get(userProfile, 'bidder_types') || null;
+  const bidderType = 'cusp';
```
Comment (M00staff) [inline src/Components/BidderPortfolio/BidderPortfolioStatRow/BidderPortfolioStatRow.jsx:32]: hardcoding to demo - this is for the CDO D3 stuff

**Comment by M00staff** [comment]
There's a lot left to be desired - could probably look better and the redux stuff needs to be cleaned up a little bit, but we can just do that part during the integration 🤷

image

**Review by M00staff — APPROVED** [review]

**Pull request merged**